### DISCUSSION TITLE
Creating Special Notes Directory

### DISCUSSION THREAD

**Discussion by DennisTheMenace780** [nvim-telekasten/telekasten.nvim#202] [General]
I am trying to use Telekasten for both personal and work related things, but I want to create a separate directory for `work` that can house my `work_templated` notes. Much like the `weekly` and `daily` notes that go to a special directory, I am curious if it is possible to create more special directories? That way, all I need to do is simply add my work directory to `.gitignore` so I can safely upload my notes to Github without worry about leaking any information.

**Comment by lambtho12** [comment]
You can create vaults and clearly separate work and personal stuff. Vaults can be in completely different paths, they do not need to have a common hierarchy. You could even go as far as creating different vaults for different clients/projects if that fits your workflow. You can then use the `switch_vaults` function to call a picker and easily swap between the different vaults.

----
As a side note, you may also be interested in [git-crypt](https://github.com/AGWA/git-crypt), which is a small tool that transparently encrypts and decrypts some files or sub-directories of your repo.  With that, you can still have your config online but the sensitive stuff is always encrypted.

**Reply by DennisTheMenace780** [reply]
Hmm, I can certainly experiment with the idea of vaults. I was hoping to try and keep everything in the same hierarchy though because I use Telekasten to also build up technical notes on whatever I need to. You can imagine maybe I learn something about GraphQL in the process of my work, but I may want to just extract this as a general thing itself. 

Because my personal and work related stuff has some technical cross over, having everything together would be helpful instead of switching back and forth. But, I can try with the vaults first to see if that workflow can be beneficial for me. Though I wonder if it would be helpful to have a feature that extends the idea of "special directories"?

**Reply by lambtho12** [reply]
I am not sure I understand everything.

If what you simply want is to have work-related notes in a separate subdirectory and add this to the gitignore, you can do it right away.
That would mean your zettelkasten would be structured like this:

```
Zettelkasten
├── personal
│   ├── personote.md
│   ├── personote.md
├── work
│   ├── secretworknote.md
│   ├── secretworknote.md
├── img
├── index.md
├── journal
├── templates
```

and then you can simply add `Zettelkasten/work` to your `.gitignore`. With a setup like this one, you can have all in one place, make as many links as you want between personal and work notes and the `gitignore` will ensure no sensitive stuff are synced.

**Reply by lambtho12** [reply]
Telekasten does not enforce a flat hierarchy. You can do as many sub-directories as you want. I do not see what would be the difference with what you call "special directories".

**Reply by DennisTheMenace780** [reply]
Sorry, when I say "special directory" I am referencing the phrasing used in the provided configuration file:

```lua
    -- dir names for special notes (absolute path or subdir name)
    dailies      = home .. '/' .. 'daily',
    weeklies     = home .. '/' .. 'weekly',
    templates    = home .. '/' .. 'templates',
```
The way these work is that when you create a daily/weekly note for instance it automatically gets added to those repositories. So what I was thinking was having something like this instead:

```lua
    -- dir names for special notes (absolute path or subdir name)
    dailies      = home .. '/' .. 'daily',
    weeklies     = home .. '/' .. 'weekly',
    templates    = home .. '/' .. 'templates',
   work            = home .. '/' .. 'work-dailies'
```

That way if I were to use a `work-dailies` templated note for example, that note would be automatically created in the specific directory (work-dailies) that I want it to be in. I did try the vault suggestion and modified the configuration settings which worked, but felt a bit awkward because of the transition necessary using the `switch_vault` command, and also not necessarily knowing on my status line (Lualine) what the current vault is (though maybe this is something I can modify, I just don't know). 

Basically if there was a way in a single vault to create a "special directory" like the dailies and weeklies, I think it would improve the UX of creating notes. But, the vault approach is also a viable alternative.

**Reply by lambtho12** [reply]
I don't really like the idea of adding a million different options for "special directories". Especially when these directories do not bring anything specific. As I understand it, it is just a regular subdirectory but for your work suff.
You can already create a note in a specific subdirectory by just calling the `create_note` function and using `work/myworknote.md` as your title. For me this is not a big deal.

What we could do to make things 'faster' is allowing some options to the lua functions used to create and search the notes so you could craft custom mappings that fit your needs.
For instance something like `:lua require('telekasten').new_note({ subdir='work' })` could create a note in the subdirectory `work` automatically and you would only have to type the title `myworknote.md`. Likewise, we could have `find_files` with specific options to only search for files in some subdirectories. You can then just have mappings like `<leader>zn` for new notes and `<leader>znw` for the same function but with the appropriate argument.

Would that resolve what you ask here?

**Reply by lambtho12** [reply]
In a more general way, I think it would be best to expose as many options as possible for the lua functions so people can craft custom mappings like that for many different things. This will simplify the codebase and make it easier to maintain for us. There is as may workflows as there are people using telekasten, implementing custom stuff ourselves will just be a mess.

**Reply by DennisTheMenace780** [reply]
> For instance something like `:lua require('telekasten').new_note({ subdir='work' })` could create a note in the subdirectory `work` automatically and you would only have to type the title `myworknote.md`. Likewise, we could have `find_files` with specific options to only search for files in some subdirectories. You can then just have mappings like `<leader>zn` for new notes and `<leader>znw` for the same function but with the appropriate argument.

I think this would be right on the nose for what I am looking for! If you would like, I can open a separate issue for a feature request and link this conversation?

**Reply by Zethtren** [reply]
I like this idea so I can more easily key bind specific directories. 

It would be nice to have all of these options available in new_note

title = '{{uuid-title}}' -- If no title then the hyphen is omitted
prompt_title = false -- Create note using uuid as filename
dir = 'path/to/subdir'
template = 'path/to/template'

Its nice having these in the config but it is very restrictive.

I would like to put all of my "fleeting" notes in a subdir so my home dir can ba more of a "TOC directory"

I would also like to have custom dirs for larger notes such as "questions", "work" So I can better organize things. 

I know you can use the built in search to find specific items super easy and that works great. But sometimes it helps to be able to just scroll through things in groups so I can decide was is still important and what needs updating without having to remember every small item.

I tried something like this:
vim.keymap.set("n", "<leader>zn", "<cmd>Telekasten new_note\<CR\>scratch/"..os.date(format)..'.md')

and it worked intermittenly.
Sometimes it went without fail but most of the times after verifying the file name and hitting enter telescope shows as empty. So I can't locate the file to edit.
